ngraph/test/type_prop/selu.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/type_prop.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(type_prop, selu_basic_inference_f32_3D)
+{
+    const auto param = make_shared<op::Parameter>(element::f32, Shape{1, 32, 32});
+    const auto alpha = make_shared<op::Parameter>(element::f32, Shape{1});
+    const auto lambda = make_shared<op::Parameter>(element::f32, Shape{1});
+    const auto selu = make_shared<op::Selu>(param, alpha, lambda);
+
+    ASSERT_EQ(selu->get_element_type(), element::f32);
+    ASSERT_EQ(selu->get_shape(), (Shape{1, 32, 32}));
+}
+
+TEST(type_prop, selu_basic_inference_f16_3D)
+{
+    const auto param = make_shared<op::Parameter>(element::f16, Shape{1, 32, 32});
+    const auto alpha = make_shared<op::Parameter>(element::f16, Shape{1});
+    const auto lambda = make_shared<op::Parameter>(element::f16, Shape{1});
+    const auto selu = make_shared<op::Selu>(param, alpha, lambda);
+
+    ASSERT_EQ(selu->get_element_type(), element::f16);
+    ASSERT_EQ(selu->get_shape(), (Shape{1, 32, 32}));
+}
+
+TEST(type_prop, selu_basic_inference_f32_5D)
+{
+    const auto param = make_shared<op::Parameter>(element::f32, Shape{12, 135, 221, 31, 15});
+    const auto alpha = make_shared<op::Parameter>(element::f32, Shape{1});
+    const auto lambda = make_shared<op::Parameter>(element::f32, Shape{1});
+    const auto selu = make_shared<op::Selu>(param, alpha, lambda);
+
+    ASSERT_EQ(selu->get_element_type(), element::f32);
+    ASSERT_EQ(selu->get_shape(), (Shape{12, 135, 221, 31, 15}));
+}
+
+TEST(type_prop, selu_basic_inference_f16_5D)
+{
+    const auto param = make_shared<op::Parameter>(element::f16, Shape{12, 135, 221, 31, 15});
+    const auto alpha = make_shared<op::Parameter>(element::f16, Shape{1});
+    const auto lambda = make_shared<op::Parameter>(element::f16, Shape{1});
+    const auto selu = make_shared<op::Selu>(param, alpha, lambda);
+
+    ASSERT_EQ(selu->get_element_type(), element::f16);
+    ASSERT_EQ(selu->get_shape(), (Shape{12, 135, 221, 31, 15}));
+}
+
+TEST(type_prop, selu_incompatible_input_type_boolean)
```

Comment:
`validate_and_infer_types` method throws two different type of exceptions, I would recommend to use try-catch together with EXPECT_HAS_SUBSTRING macro to ensure the exception thrown is the one expected. Example [here](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/type_prop/bucketize.cpp#L137)